src/Akka.Persistence.MongoDb/Journal/MongoDbJournal.cs

```diff
@@ -518,7 +520,8 @@ protected virtual async Task<long> ReplayAllEventsAsync(ReplayAllEvents replay)
 
             var readFilter = filters.Any()? builder.And(filters): null;
 
-            if(readFilter is null)
+            //If filters is empty, this will deadlock - so when readFilter is null find all
+            if (readFilter is null)
```

Comment:
The code inside both if blocks is identical, it would be better if you change the `readFilter` to a `Func` that always returns true instead of duplicating the whole LINQ code block